src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java

```diff
@@ -675,6 +682,18 @@ public void notifyContentEvent(@ValidateStringParam String site, @ValidateSecure
         applicationContext.publishEvent(new ContentEvent(securityService.getAuthentication(), site, path));
     }
 
+    @Override
+    public boolean validateAndCreateFolder(@SiteId String site, String path, String name)
+            throws ServiceLayerException, UserNotFoundException, ValidationException {
+        siteService.checkSiteExists(site);
```

Comment:
@phuongnq please consider these scenarios as well: https://github.com/craftercms/craftercms/issues/7047

Also, please take a look at the other API V1 endpoints and implement a similar approach as needed

Reply:
Updated to support curly braces.
Also, update similar APIs to validate the writing path:

```
rename-folder
write-content
```